frontend/src/components/NavBar.tsx

```diff
@@ -188,89 +107,16 @@ export default function NavBar({
-                    </Button>
-                    <Button
-                      colorScheme="blue"
-                      onClick={timerControls.playPauseToggle}
-                    >
-                      <Icon as={isLive ? FaPause : FaPlay} />
-                    </Button>
-                    <Button
-                      colorScheme="blue"
-                      onClick={() => {
-                        performSongNavAction("next");
-                      }}
-                    >
-                      <Icon as={FaFastForward} />
-                    </Button>
-                  </Flex>
-                  <Flex justifyContent="space-between" mx="1rem" my=".5rem">
-                    <Button
-                      colorScheme="gray"
-                      onClick={() => {
-                        resetAppState();
-                        timerControls.refresh();
-                      }}
-                    >
-                      <Icon as={FaUndoAlt} />
-                    </Button>
-                    <Button
-                      colorScheme="gray"
-                      onClick={() => performSongNavAction("delete")}
-                    >
-                      <Icon as={FaTrash} />
-                    </Button>
-                    <Button
-                      colorScheme="gray"
-                      onClick={() => {
-                        if (!document.fullscreenElement) {
-                          document.body.requestFullscreen();
-                        } else {
-                          document.exitFullscreen();
-                        }
-                      }}
-                    >
-                      <Icon as={FaExpandAlt} />
-                    </Button>
-                  </Flex>
-                </Flex>
-              )}
-              <RouterLink to="../live/">
-                <MenuItem icon={<Icon as={FaHome} />}>Home</MenuItem>
-              </RouterLink>
-              {asyncSongbook?.result?.data?.is_noodle_mode && (
-                <RouterLink to="list">
-                  <MenuItem icon={<Icon as={GrUnorderedList} />}>
-                    Song List
-                  </MenuItem>
-                </RouterLink>
-              )}
-              <MenuItem
-                icon={colorMode === "light" ? <MoonIcon /> : <SunIcon />}
-                onClick={toggleColorMode}
-              >
-                {colorMode === "light" ? "Night Mode" : "Day Mode"}
-              </MenuItem>
-            </MenuList>
-          </Menu>
+          <HamburgerMenu
+            isSongbookOwner={isSongbookOwner}
+            isMobileDevice={isMobileDevice}
+            timerControls={timerControls}
+            isLive={isLive}
+            setIsLive={setIsLive}
+            addSongDrawerOutlet={addSongDrawerOutlet}
+            asyncSongbook={asyncSongbook}
+            resetAppState={resetAppState}
+          />
         </Flex>
 
         {!isMobileDevice && (
```

Comment:
Feels much better to cut down the size of this file!

Reply:
I'm wondering about the viability of pulling the nav part(s) of the hamburger menu out into a separate component. That would enable us to construct it as needed for each various page we're using it on.